paywall/src/utils/polling.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * poll at a regular interval, with the ability to pause polling or end it
+ */
+export default function pollWithConditions(
+  pollingCallback,
+  intervalMs,
+  disablePolling,
+  handleError = () => {}
+) {
+  async function poll() {
+    try {
+      if (disablePolling()) return
+      await pollingCallback()
```

Comment:
This implied that the polling callback must be returning a promise. Is that assuming too much? I worry about coupling here.